subgraphs/_reference_/src/sdk/protocols/generic/pool.ts

```diff
@@ -0,0 +1,387 @@
+import { Bytes, BigDecimal, BigInt, Address } from "@graphprotocol/graph-ts";
+import { Pool as PoolSchema, Token } from "../../../../generated/schema";
+import { ProtocolManager } from "./protocol";
+import {
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  RewardTokenType,
+} from "../../util/constants";
+import { exponentToBigDecimal } from "../../util/numbers";
+import {
+  sortArrayByReference,
+  sortBytesArray,
+  updateArrayAtIndex,
+} from "../../util/arrays";
+import { TokenManager } from "./tokens";
+import { PoolSnapshot } from "./poolSnapshot";
+import { SDK } from ".";
+import { CustomEventType } from "../../util/events";
+
+type onCreatePoolCallback<T> = (
```

Comment:
I liked this approach of having a callback that runs only when a pool is created and to which you can pass custom arguments, but it has been very confusing for the people who have worked with it so far.

I’m inclined to remove it if we don’t find a way to make it more straightforward. What do you think?